packages/calcite-components/src/components/date-picker-month-header/date-picker-month-header.tsx

```diff
@@ -323,16 +408,127 @@ export class DatePickerMonthHeader {
     commit?: boolean;
     offset?: number;
   }): void {
-    const { yearInput, activeDate } = this;
+    const { yearInputEl, activeDate } = this;
     const inRangeDate = this.getInRangeDate({ localizedYear, offset });
 
     // if you've supplied a year and it's in range, update active date
     if (inRangeDate) {
-      this.calciteInternalDatePickerSelect.emit(inRangeDate);
+      this.calciteInternalDatePickerMonthHeaderSelect.emit(inRangeDate);
     }
 
     if (commit) {
-      yearInput.value = this.formatCalendarYear((inRangeDate || activeDate).getFullYear());
+      yearInputEl.value = this.formatCalendarYear((inRangeDate || activeDate).getFullYear());
+    }
+  }
+
+  private async setSelectMenuIconOffset(select: HTMLCalciteSelectElement): Promise<void> {
+    const iconEl = select.shadowRoot.querySelector("calcite-icon");
```

Comment:
FIXME: let's avoid querying shadow roots.

Reply:
This is required to set dynamic width for month select.

Reply:
Could we get the icon container dimensions from the design spec and tokens? If not, I’m inclined to hardcode the sizes (based on the spec) while we explore alternatives in a follow-up issue.

Reply:
Initial idea is to use hardcoded values but it can be brittle if the layout of select or sizing of icon changes. Prefer using tokens but they aren't defined for the `inline-size` & `block-size` of icon yet. 

Looking at it now, i realized we can get rid of all the querySelectors since we use only `scale="m"` for `select` in `date-picker-month-header` & `icon` size is fixed. Additionally, `tokens` for spacing can be used to calibrate offset width.